Build fails on macOS Sierra (a2x, xsltproc)

Here is a [full log](https://gist.github.com/elzii/aa32a15a9bf4978c615ea172fa6c1491) from brew

Relevant bit (I think):

```
a2x --no-xmllint -f manpage ../doc/kak.1.txt
a2x --no-xmllint -f manpage ../doc/manpages/commands.asciidoc
a2x --no-xmllint -f manpage ../doc/manpages/execeval.asciidoc
a2x --no-xmllint -f manpage ../doc/manpages/expansions.asciidoc
a2x --no-xmllint -f manpage ../doc/manpages/faces.asciidoc
a2x --no-xmllint -f manpage ../doc/manpages/faq.asciidoc
a2x --no-xmllint -f manpage ../doc/manpages/highlighters.asciidoc
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/usr/local/Cellar/asciidoc/8.6.9_1/etc/asciidoc/docbook-xsl/manpage.xsl" "/private/tmp/kakoune-20161124-72283-jl9yhi/doc/manpages/faces.xml" returned non-zero exit status 5
make: *** [../doc/manpages/faces.gz] Error 1
make: *** Waiting for unfinished jobs....
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/usr/local/Cellar/asciidoc/8.6.9_1/etc/asciidoc/docbook-xsl/manpage.xsl" "/private/tmp/kakoune-20161124-72283-jl9yhi/doc/kak.1.xml" returned non-zero exit status 5
make: *** [../doc/kak.1.gz] Error 1
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/usr/local/Cellar/asciidoc/8.6.9_1/etc/asciidoc/docbook-xsl/manpage.xsl" "/private/tmp/kakoune-20161124-72283-jl9yhi/doc/manpages/execeval.xml" returned non-zero exit status 5
make: *** [../doc/manpages/execeval.gz] Error 1
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/usr/local/Cellar/asciidoc/8.6.9_1/etc/asciidoc/docbook-xsl/manpage.xsl" "/private/tmp/kakoune-20161124-72283-jl9yhi/doc/manpages/faq.xml" returned non-zero exit status 5
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/usr/local/Cellar/asciidoc/8.6.9_1/etc/asciidoc/docbook-xsl/manpage.xsl" "/private/tmp/kakoune-20161124-72283-jl9yhi/doc/manpages/expansions.xml" returned non-zero exit status 5
make: *** [../doc/manpages/faq.gz] Error 1
make: *** [../doc/manpages/expansions.gz] Error 1
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/usr/local/Cellar/asciidoc/8.6.9_1/etc/asciidoc/docbook-xsl/manpage.xsl" "/private/tmp/kakoune-20161124-72283-jl9yhi/doc/manpages/highlighters.xml" returned non-zero exit status 5
make: *** [../doc/manpages/highlighters.gz] Error 1
a2x: ERROR: "xsltproc"  --stringparam callout.graphics 0 --stringparam navig.graphics 0 --stringparam admon.textlabel 1 --stringparam admon.graphics 0  "/usr/local/Cellar/asciidoc/8.6.9_1/etc/asciidoc/docbook-xsl/manpage.xsl" "/private/tmp/kakoune-20161124-72283-jl9yhi/doc/manpages/commands.xml" returned non-zero exit status 5
make: *** [../doc/manpages/commands.gz] Error 1
```

As per #579 here are some of the outputs requested:

```
~ which a2x
/usr/local/bin/a2x
```
```
~ which a2x.py
/usr/local/bin/a2x.py
```
```
~ make -v
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
This program built for i386-apple-darwin11.3.0
```
```
~ printf 'all:\n\techo $(shell which a2x)' | make -f -
echo /usr/local/bin/a2x
/usr/local/bin/a2x
```
```
~ printf 'all:\n\techo $(shell which a2x.py)' | make -f -
echo /usr/local/bin/a2x.py
/usr/local/bin/a2x.py
```


```
~ printf 'all:\n\techo $$PATH' | make -f -
/Users/elzi/.rvm/bin:
/Users/elzi/.rvm/gems/ruby-2.3.1/bin:
/Users/elzi/.rvm/gems/ruby-2.3.1@global/bin:
/Users/elzi/.rvm/rubies/ruby-2.3.1/bin:
/Users/elzi/.nvm/versions/node/v5.0.0/bin:
/usr/local/bin:
/usr/bin:
/bin:
/usr/sbin:
/sbin:
/opt/X11/bin:
/usr/local/go/bin:
/usr/local/bin:
/usr/bin:
/usr/sbin:
/usr/local/sbin:
```

--- a/contrib/kakoune.rb
+++ b/contrib/kakoune.rb
@@ -5,7 +5,8 @@ class Kakoune < Formula
   head "https://github.com/mawww/kakoune.git"
 
   depends_on 'boost'
-  depends_on 'asciidoc' => :build
+  depends_on 'docbook-xsl' => :build
+  depends_on 'asciidoc' => [:build, 'with-docbook-xsl']
 
   def install
     cd 'src' do

